clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3649,7 +3652,7 @@ public DeleteConsumerGroupOffsetsResult deleteConsumerGroupOffsets(
 
     @Override
     public Map<MetricName, ? extends Metric> metrics() {
-        return Collections.unmodifiableMap(this.metrics.metrics());
+        return new HashMap<>(this.metrics.metrics());
```

Comment:
We should keep it immutable, since other client components, such as consumer and producer, return immutable collection